File: Arrays/Techniques/two_pointer.java
package Techniques;
public class two_pointer {

    public static void main(String[] args) {

        int[] arr = {1, 2, 3, 4, 6, 8};

        int target = 10;

        int left = 0;
        int right = arr.length - 1;

        while (left < right) {

            int sum = arr[left] + arr[right];

            if (sum == target) {
                System.out.println(
                    "Pair found: " + arr[left] + " + "
                    + arr[right] + " = " + target
                );
                return;
            }

            if (sum < target) {
                left++;
            } else {
                right--;
            }
        }

        System.out.println("Pair not found");
    }
}

File: Arrays/Techniques/two_pointer.c
#include <stdio.h>

int main() {

    int arr[] = {1, 2, 3, 4, 6, 8};
    int n = 6;

    int target = 10;

    int left = 0;
    int right = n - 1;

    while (left < right) {

        int sum = arr[left] + arr[right];

        if (sum == target) {
            printf("Pair found: %d + %d = %d\n",
                   arr[left], arr[right], target);
            return 0;
        }

        if (sum < target) {
            left++;
        } else {
            right--;
        }
    }

    printf("Pair not found\n");

    return 0;
}